dask/array/tests/test_reductions.py

```diff
@@ -566,3 +566,25 @@ def test_regres_3940(func):
     assert func(a).name != func(a + 1).name
     assert func(a, axis=0).name != func(a).name
     assert func(a, axis=0).name != func(a, axis=1).name
+
+
+def test_trace():
+    def _assert_eq(a, b, dtype=np.int64):
+        return assert_eq(a.compute(), np.asarray(b, dtype))
+
+    a = da.from_array(np.arange(12).reshape((3, 4)), 1)
+    _assert_eq(a.trace(), 15)
```

Comment:
Rather than include the correct result, I recommend just relying on NumPy here.  That way we don't accidentally introduce bugs in our tests :)

So for example maybe something like this:

```python
a = np.arange(12).reshape((3, 4))
b = da.from_array(a, chunks=(1, 1))

assert_eq(a.trace(), b.trace())
assert_eq(a.trace(0), b.trace(0))
...
```